src/fheroes2/game/game_scenarioinfo.cpp

```diff
@@ -282,25 +315,31 @@ namespace
             // click select
             if ( HotKeyPressEvent( Game::HotKeyEvent::MAIN_MENU_SELECT_MAP ) || le.MouseClickLeft( buttonSelectMaps.area() ) ) {
                 const Maps::FileInfo * fi = Dialog::SelectScenario( lists );
+                const std::string currentMapName = conf.getCurrentMapInfo().filename;
 
-                if ( fi ) {
-                    Game::SavePlayers( conf.getCurrentMapInfo().filename, conf.GetPlayers() );
+                if ( fi && fi->filename != currentMapName ) {
+                    Game::SavePlayers( currentMapName, conf.GetPlayers() );
                     conf.SetCurrentFileInfo( *fi );
+
+                    mapTitleArea.restore();
+                    RedrawMapTitle( scenarioBoxRoi );
                     Game::LoadPlayers( fi->filename, players );
 
+                    opponentsArea.restore();
+                    classArea.restore();
+                    handicapArea.restore();
+                    ratingArea.restore();
+
                     updatePlayers( players, humanPlayerCount );
                     playersInfo.UpdateInfo( players, pointOpponentInfo, pointClassInfo );
 
-                    RedrawScenarioStaticInfo( rectPanel );
-                    RedrawDifficultyInfo( pointDifficultyInfo );
                     playersInfo.resetSelection();
                     playersInfo.RedrawInfo( false );
-                    ratingRoi = RedrawRatingInfo( rectPanel.getPosition(), rectPanel.width );
-                    levelCursor.setPosition( coordDifficulty[Game::getDifficulty()].x, coordDifficulty[Game::getDifficulty()].y ); // From 0 to 4, see: Difficulty enum
-                    buttonOk.draw();
-                    buttonCancel.draw();
-                }
 
+                    ratingRoi = RedrawRatingInfo( roi.getPosition(), roi.width );
+                    levelCursor.setPosition( coordDifficulty[Game::getDifficulty()].x - levelCursorOffset,
+                                             coordDifficulty[Game::getDifficulty()].y - levelCursorOffset ); // From 0 to 4, see: Difficulty enum
+                }
                 display.render();
```

Comment:
Here we can do render only in the `activeArea()`:
```suggestion
                display.render( roi );
```

Reply:
Ow, then its OK. :)